packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -133,6 +141,41 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     [apiRef, rowId],
   );
 
+  const publishClick = React.useCallback(
+    (event) => {
+      const cell = findParentElementFromClassName(event.target, gridClasses.cell);
+      const field = cell?.getAttribute('data-field');
+
+      // Check if the field is available because the cell that fills the empty
+      // space of the row has no field.
+      if (field) {
+        // User clicked in the checkbox added by checkboxSelection
+        if (field === GRID_CHECKBOX_SELECTION_COL_DEF.field) {
+          return;
+        }
+
+        // User opened a detail panel
+        if (field === '__detail_panel_toggle__') {
```

Comment:
Not for this PR but should we have a file with all the const in `data-grid` folder, including the pro const?